sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java

```diff
@@ -246,7 +246,7 @@ public void decodeSingleDictionaryId(
     }
   }
 
-  private static class UnsignedIntegerUpdater implements ParquetVectorUpdater {
+  private static class LongIntegerUpdater implements ParquetVectorUpdater {
     @Override
     public void readValues(
```

Comment:
readValues() and readValue() are now inconsistent, the former is calling readUnsignedIntegers() which still use Integer.toUnsignedLong() internally.

Instead of touching the UnsignedIntegerUpdater class, it appears that your intention above in this file is to use LongUpdater instead.

Reply:
right. Not sure whether a new class make sense for clarity and future merge. The file is recent by "sunchao@apple.com",maybe you can explicitly @ in SPARK-36990 it not yet for attention.

Reply:
Yeah seems this behavior is a well known problem: SPARK-35461
Fixed the readValues as well